rmw_fastrtps_shared_cpp/src/rmw_node_info_and_types.cpp

```diff
@@ -59,7 +59,7 @@ rmw_ret_t __get_guid_by_name(
   const char * node_namespace, GUID_t & guid)
 {
   auto impl = static_cast<CustomParticipantInfo *>(node->data);
-  if (strcmp(node->name, node_name) == 0) {
+  if (strcmp(node->name, node_name) == 0 && strcmp(node->namespace_, node_namespace) == 0) {
```

Comment:
Realized of this preexisting bug when uncommenting some tests in `rcl` `test_graph.cpp`.